feat(leadspace): introduced leadspace medium variant

andysherman2121 opened the pull request:
### Related Ticket(s)

#5231 

### Description

Added `medium` and `tall` `size` prop


<!-- React and Web Component deploy previews are enabled by default. -->
<!-- To enable additional available deploy previews, apply the following -->
<!-- labels for the corresponding package: -->
<!-- *** "package: services": Services -->
<!-- *** "package: utilities": Utilities -->
<!-- *** "package: styles": Carbon Expressive -->
<!-- *** "RTL": React / Web Components (RTL) -->
<!-- *** "feature flag": React / Web Components (experimental) -->

jeffchew commented:
@andysherman2121 looks like a bunch of build issues. Would you be able to fix today?

andysherman2121 commented:
> @andysherman2121 looks like a bunch of build issues. Would you be able to fix today?

@jeffchew Yup! i've been working through them

andysherman2121 commented:
@DragosRistici Deploy preview is up!

asudoh commented on packages/web-components/src/components/leadspace/leadspace.ts:
```diff
@@ -59,20 +59,50 @@ class DDSLeadSpace extends StableSelectorMixin(LitElement) {
+   */
+  protected _getContainerClass() {
+    return classMap({
+      [`${prefix}--leadspace__container-medium`]: this.size === LEADSPACE_SIZE.MEDIUM,
```
BEM uses two dashes for variants:

```suggestion
      [`${prefix}--leadspace__container--medium`]: this.size === LEADSPACE_SIZE.MEDIUM,
```

asudoh reviewed (commented):
Thank you for the update @andysherman2121!

jeffchew commented:
@andysherman2121 instead of a knob, can it be a separate story? Probably two more, with and without image.

@shixiedesign I'm not sure if we wanted to remove the small variants in this PR or not? I think I recall discussing that but couldn't find that in any issues.

jeffchew commented:
The documentation for `type` is outdated, can you update to include `medium`?

Also, the stories are a bit confusing, Wondering if we should rename the stories and have the available options as `tall` (not default), `medium`, `centered` (and not sure if we're removing `small`, @shixiedesign ?)

<img width="1080" alt="Screen Shot 2021-03-11 at 6 38 10 PM" src="https://user-images.githubusercontent.com/1641214/110869707-fad78400-8298-11eb-9d94-79df186d2d90.png">

andysherman2121 commented:
> @andysherman2121 instead of a knob, can it be a separate story? Probably two more, with and without image.
> 
> @shixiedesign I'm not sure if we wanted to remove the small variants in this PR or not? I think I recall discussing that but couldn't find that in any issues.

@jeffchew James said that it should be a `knob`. I can definitely make it into a story though if that's what we want instead

jeffchew commented:
> @jeffchew James said that it should be a `knob`. I can definitely make it into a story though if that's what we want instead

I think we should, but see my comment above (I changed my comment a bit). My goal is more about trying to capture the visual regression coverage with Percy.

shixiedesign commented:
Until @DragosRistici wakes up, here's a few things I noticed: 
- [ ] Buttons & copy are not lower left pinned. 
Currently: 
![image](https://user-images.githubusercontent.com/15144993/110885447-dbe6eb00-82b4-11eb-912b-20767e8034c1.png)
Expecting:
![image](https://user-images.githubusercontent.com/15144993/110885575-1cdeff80-82b5-11eb-93f2-0eac9e06f949.png)

- [ ] Probably related to above, when `sm` breakpoint and line wraps, buttons gets pushed out of lead space. Expecting buttons and small copy to be bottom pinned 
![Mar-11-2021 22-03-09](https://user-images.githubusercontent.com/15144993/110885873-9ecf2880-82b5-11eb-9bd8-80005132a401.gif)

andysherman2121 commented:
@shixiedesign Ah you're right i'll fix that

andysherman2121 commented:
@shixiedesign Updated! The copy and buttons should be pinned to the bottom left now

DragosRistici commented:
First of all, thank you so much, Andy @andysherman2121 , for your fast turnaround. And thank you to Shixie @shixiedesign  for getting to it first.


I’d like to start with a couple of small general tweaks.  

1. Because we are showcasing a white lead space (no image) on a white background page, I suggest adding a Gray10 background color to the bx--leadspace__overlay div, so the adopter can “see” the lead space, as we are giving him height options with quite significant differences. Or at least add a border-bottom of 1px solid gray30. (Of course, if the gradient wouldn't be there when there's no image, that would be awesome, but i think its fine with the gray10 behind it.)
<img width="1031" alt="Screenshot 2021-03-12 at 15 19 19" src="https://user-images.githubusercontent.com/48327647/110945456-60357e80-8346-11eb-8fa0-92d31d8c265b.png">


2. As the design specs say, we should use this opportunity to showcase it as close as possible to the real world, that’s why we added the text “Heading could go on two lines max” because headings usually go on more than one row. 



///////////////

DESKTOP BREAK POINTS (max, xlg, lg)


3. Display-01’ has a bottom spacing of 6rem, like in the image below. And I would like to use this occasion to ask you guys to use margin to set the distance between elements, because if you use padding, you’ll just make the element bigger and the padding space will get the styling if case may be, too.    It could be like this:
<img width="1031" alt="Screenshot 2021-03-12 at 13 22 48" src="https://user-images.githubusercontent.com/48327647/110946188-4183b780-8347-11eb-8d02-7c61607234ec.png">

because the design specs are like this:
![image](https://user-images.githubusercontent.com/48327647/110946629-c40c7700-8347-11eb-9b18-a06a01d07f39.png)




4. Expressive-heading-03 should have no top spacing, but a bottom spacing of 2rem;  


5. Button group should have no extra spacing top/bottom. But the Dds-button-group-item has a margin-top of 1rem. If the component has it already, then maybe you cut it from the above element, so in the end we get A TOTAL OF 2rem between the expressive-heading-03 and the dds-button-group element.  

Attention: Please, make sure that the margin-top of 1 rem of the button item we have does not get used in other components like that. If that’s the case, then for the lead space reduce it from the above element.
![image](https://user-images.githubusercontent.com/48327647/110947028-4eed7180-8348-11eb-8338-300ac8ecd770.png)




///////////////

TABLET BREAK POINT (md)

<img width="400" alt="Screenshot 2021-03-12 at 14 46 43" src="https://user-images.githubusercontent.com/48327647/110947404-c7543280-8348-11eb-98b0-ac84ea586f83.png"><img width="400" alt="Screenshot 2021-03-12 at 14 47 01" src="https://user-images.githubusercontent.com/48327647/110947539-f5d20d80-8348-11eb-9d92-9f291f573841.png">


Please follow the specs for this one. The image is above the rest, just like the mobile version. If you need to see another example, please check the ibm.com homepage as they are using the same visual approach for the md break point. 

<img alt="Screenshot 2021-03-12 at 15 14 05" src="https://user-images.githubusercontent.com/48327647/110947925-6bd67480-8349-11eb-8e89-f7aae6982a36.png"><img width="400" alt="Screenshot 2021-03-12 at 15 42 20" src="https://user-images.githubusercontent.com/48327647/110948069-945e6e80-8349-11eb-8490-ef8ae42b7e9a.png"><img width="400" alt="Screenshot 2021-03-12 at 15 13 58" src="https://user-images.githubusercontent.com/48327647/110947940-6ed16500-8349-11eb-99b3-1f1eb41fdee8.png">
<img alt="Screenshot 2021-03-12 at 15 13 42" src="https://user-images.githubusercontent.com/48327647/110947942-70029200-8349-11eb-8357-1d18322cd5f6.png">



///////////////

MOBILE BREAK POINT (sm)

Also, for the mobile, please take the same approach. There is no lead space height for the md and sm break points. Here the elements have precise spacing between them, exactly like in the specs. If there is an image, it will not be as background. 

<img width="629" alt="Screenshot 2021-03-12 at 15 10 08" src="https://user-images.githubusercontent.com/48327647/110948398-f7e89c00-8349-11eb-8f9e-b6a17fa99191.png"><img width="318" alt="Screenshot 2021-03-12 at 15 50 43" src="https://user-images.githubusercontent.com/48327647/110949002-c0c6ba80-834a-11eb-9b46-77c7a55ed7d1.png">
<img  alt="Screenshot 2021-03-12 at 15 09 15" src="https://user-images.githubusercontent.com/48327647/110948415-fcad5000-8349-11eb-9c57-8c1a33394e8e.png">

Thank you so much for your effort and let me know if you have any questions. Thanks.

andysherman2121 commented:
@DragosRistici Thanks for the comments! I talked to @photodow and he mentioned that for this story we weren't including the image functionality?

DragosRistici commented:
@andysherman2121 I didn't know that, but if he says so, then it must be true. One last thing, please, about the gradient. Look at the specs, there are 3 points of contact for it: 0, 1/2 and 3/4. They have the opacity set to 1, .4 and 0, respectively. Right now, there are 2, and one is .95 and the other one is 0 at 3/4. Thanks again, Andy. 


<img width="1284" alt="Screenshot 2021-03-12 at 16 12 00" src="https://user-images.githubusercontent.com/48327647/110951436-b8bc4a00-834d-11eb-9a6b-fab11960e283.png">


<img width="525" alt="Screenshot 2021-03-12 at 16 09 26" src="https://user-images.githubusercontent.com/48327647/110951251-87dc1500-834d-11eb-89ed-336ee3e8d561.png">
<img width="532" alt="Screenshot 2021-03-12 at 16 09 21" src="https://user-images.githubusercontent.com/48327647/110951256-8a3e6f00-834d-11eb-818e-4e3e53495a6c.png">
<img width="429" alt="Screenshot 2021-03-12 at 16 09 07" src="https://user-images.githubusercontent.com/48327647/110951258-8ad70580-834d-11eb-96f8-4dfca53f8a4e.png">

andysherman2121 commented:
Hi @DragosRistici !  

I've added the `gray-10` background and fixed the gradient. I couldn't get rid of the spacing above the button-group so I took  your advice and changed the bottom spacing on the description text.

DragosRistici commented:
@andysherman2121 Thank you, Andy. You are a master! I rest my case. Its fine by me. Now what? :)

DragosRistici reviewed (approved):
Thank you.

IgnacioBecerra reviewed (approved):
LGTM, good job!
